### DISCUSSION TITLE
Is the server performance affecting the clients performance?

### DISCUSSION THREAD

**Discussion by juangea** [rustdesk/rustdesk#5196] [Q&A] [answered]
I have the rustdesk server in a NAS machine, it's not very powerful, but it seems to work well, however I have a MacMini server too, does the machine performance affects the quality / framerate of the clients? 

The clients are very powerful machines, with 3090 as GPUs and plenty of ram and cpu power, however I can only see 25Fps over gigabit LAN

**Comment by SergeyMy** [comment]
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/9c8bf5c3-756e-4e5b-8517-465309edc0ab)
+on server
export LIMIT_SPEED=**
export SINGLE_BANDWIDTH=**

**Reply by juangea** [reply]
I have the server with the docker image, any way I can configure that for the docker image?
I h lave the client side configured that way :)

**Reply by SergeyMy** [reply]
https://rustdesk.com/docs/en/self-host/install/
i think i need to add
in **docker-compose.yml**
to the section _command: LIMIT_SPEED=** hbbr ...._

**Reply by juangea** [reply]
So I cannot do it right now?

Do I have to wait for an update or is there an specific line I have to put in the yml?

**Reply by juangea** [reply]
@SergeyMy just to be clear, in the yaml this line should be:

`command: LIMIT_SPEED=** SINGLE_BANDWIDTH=** hbbr`

And that's it?

**Reply by SagarKandel** [reply]
can you provide more info? Whats the proper way to tinker connection settings  in docker-compose.yaml file?

**Comment by xlionjuan** [comment]
My NAS is Synology DS420+ with Intel Celeron J4025, normally, the CPU usage won't over 10%.

**Comment by juangea** [comment]
Yeah, I saw that, but the problem was that I’m not getting the performance I see reported by others, and I suspect the problems are those config settings.

but I’m not sure how to apply them to the docker version, I’m executing it in a Terramaster NAS, similar to your synology.

**Reply by xlionjuan** [reply]
Have you tried host the server on your Mac mini? Will the performance problem exists?

**Reply by juangea** [reply]
How? It would be docker as in the NAS, there is no MacOS support, right?

I’ll try anyways to see if it improves :)

**Reply by xlionjuan** [reply]
But Docker has macOS version..... you can try it on macOS

https://docs.docker.com/desktop/install/mac-install/

**Reply by juangea** [reply]
Yes. I already have docker in my server, but I still don’t know how to apply those settings, I’ll try it anyways to see if performance changes :)

**Reply by juangea** [reply]
I saw no difference in here, at least nothing I can measure I'm afraid.

**Reply by xlionjuan** [reply]
Try connect though IP on same LAN, it can sure what the accurate problem is.

**Reply by juangea** [reply]
Except one of the connections, all the connection I do are on the same lan, and I tested both with the server and with direct IP connection, same reuslts.

**Reply by xlionjuan** [reply]
> Except one of the connections, all the connection I do are on the same lan, and I tested both with the server and with direct IP connection, same reuslts.

Have you tried nightly version?

**Reply by juangea** [reply]
I always use nightlies, 1.2.2 right now.

With the ENV solution I posted below it seems to have improved, I reach 60fps, which is good, I'll test increasing limits a bit more, but it has improved for sure :)

BTW only with h265, VP9 is slow for some reason.

**Comment by juangea** [comment]
@SergeyMy can you elaborate a bit on how should I apply those settings please :)

Thanks in advance!

**Reply by xlionjuan** [reply]
The method he provide isn't for docker

**Reply by xlionjuan** [reply]
https://github.com/rustdesk/rustdesk-server#env-variables

But I don't think it will make any difference 
![Screenshot_20230812-174752](https://github.com/rustdesk/rustdesk/assets/25243894/76a2574d-3b72-498a-88bf-f7982c06f6f7)

Start your HBBR and screenshot those log related "BANDWIDTH"

**Reply by juangea** [reply]
Ok, I’ll post it here later :)

**Reply by juangea** [reply]
I get the same information as you:

![image](https://github.com/rustdesk/rustdesk/assets/20524718/3e8400a6-a826-4efb-9603-94a4e00c8e29)

How can I change that limit speed and the single bandwidth?

**Reply by juangea** [reply, marked as answer]
Using this ENV:

https://github.com/rustdesk/rustdesk/discussions/5196#discussioncomment-6708040

<html>
<body>


variable | description
-- | --
LIMIT_SPEED | speed limit (in Mb/s)
TOTAL_BANDWIDTH | max total bandwidth (in Mb/s)
SINGLE_BANDWIDTH | max bandwidth for a single connection (in Mb/s)


</body>
</html>

I've managed to improve those limits to 128Mb and 256Mb respectively (limit and single) and I think it has improved, I can reach 60fps, but I don't past form that, even watching a full screen 120fps video in youtube at 1080p.

I'm not sure whats holding it back, but I'll try increasing it a bit more to see the effects, but probably this is the key :)

**Reply by SergeyMy** [reply]
well you can whenever you want

**Reply by juangea** [reply]
Thanks!

**Reply by SagarKandel** [reply]
how did you apply these in docker?
can you help?

thanks in advance
